src/api-server.ts

```diff
@@ -436,15 +445,22 @@ async function subscribeToRoom(roomName: string): Promise<any> {
             }
 
             console.log("roomName",roomName)
-            if (streamConnectionTopic.get(roomName) !== undefined) {
-                streamConnectionTopic.get(roomName).write({
-                    channelNewData: {
-                        from: message.from,
-                        data: Buffer.from(JSON.stringify(message.data)),
-                        nonce: message.seqno,
-                        channel: channels
+            if (streamConnectionTopic.get(roomName) !== undefined ) {
+                for (let key of streamConnectionTopic.get(roomName).keys()) {
+                    const payload = message.data as JsonObject
+                    if (key === payload.to) {
+                        streamConnectionTopic.get(roomName).get(key).write({
+                            channelNewData: {
+                                from: message.from,
+                                data: Buffer.from(JSON.stringify(payload.message)),
+                                nonce: message.seqno,
+                                channel: channels
+                            }
+                        });
+                        break;
                     }
-                });
+                }
+                
```

Comment:
Why are we using a for to find a map's value by its key?
I think this code be refactored in something like:

```typescript
            const payload = message.data as JsonObject
            streamConnectionTopic.get(roomName)?.get(payload.to)?.write({
                    channelNewData: {
                        from: message.from,
                        data: Buffer.from(JSON.stringify(payload.message)),
                        nonce: message.seqno,
                        channel: channels
                    }
                });
```

Reply:
refactored